R/deployApp.R

```diff
@@ -252,15 +265,27 @@ deployApp <- function(appDir = getwd(),
 
   # determine the deployment target and target account info
   recordPath <- findRecordPath(appDir, recordDir, appPrimaryDoc)
-  target <- deploymentTarget(
-    recordPath = recordPath,
-    appName = appName,
-    appTitle = appTitle,
-    appId = appId,
-    account = account,
-    server = server,
-    forceUpdate = forceUpdate
-  )
+  if (is.null(appId)) {
+    target <- deploymentTarget(
+      recordPath = recordPath,
+      appName = appName,
+      appTitle = appTitle,
+      account = account,
+      server = server,
+      forceUpdate = forceUpdate
+    )
+  } else {
+    if (!is.null(appName)) {
+      cli::cli_warn("{.arg appName} is ignored when {.arg appId} is set")
```

Comment:
@aronatkins this is not quite what you suggested, but your idea seemed like a lot of work and I think this is fairly close. I also like being able to say "if you specified `appId`, then existing deployments are ignored.